### ISSUE TITLE
Add test coverage for delete events in e2e tests

### ISSUE THREAD

**Issue by PrasadG193** [kubeshop/botkube#354]
In e2e test suite, we mock resource creation with fake client and verify the BotKube event generations.
https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/create/create.go#L46

We need to add similar test cases to verify delete event handling.
We can use [Delete](https://godoc.org/k8s.io/client-go/kubernetes/typed/core/v1#PodInterface) API for Deleting a resource.

Add test cases to validate the following behaviour:

- [ ] Validate delete events are handled for a resource with `events: [all]` in test config https://github.com/infracloudio/botkube/blob/develop/test/resource_config.yaml - @ShubhamSood1406

- [x] Validate delete events are handled with only `events: [delete]` config for a resource - @Divya063

- [x] Validate delete events are skipped when `delete` event is not configured for a resources in test config. e.g `events: [create, update, error]`- @A-kanksh-a 

- [x] Validate that delete events are skipped when an event occurs for a namespace which is not configured - @A-kanksh-a 

- [x] Validate that delete events are skipped when an event occurs for a resource which is not added in resource_config - @A-kanksh-a 

Each test case can be added as a separate PR. Multiple PR as expected for this task. 
Guidelines for contributors:
- Post comment on the issue informing others about the test case you want to add
- Mark the issue as done once the PR is merged
- Please make sure that no one else is working on the same test case to avoid duplicate PRs

**Comment by Divya063** [comment]
@PrasadG193 I would like to add test cases for the following behavior - Validate delete events are handled with only events: [delete] config for a resource

**Comment by PrasadG193** [comment]
Thanks, @Divya063 - Please let me know if you have any doubts.

**Comment by A-kanksh-a** [comment]
@PrasadG193 I would like to contribute to following behaviour

- Validate delete events are skipped when delete event is not configured for a resources in test config. e.g events: [create, update, error]
- Validate that delete events are skipped when an event occurs for a namespace which is not configured
- Validate that delete events are skipped when an event occurs for a resource which is not added in resource_config

**Comment by ShubhamSood1406** [comment]
@PrasadG193 Can I do 

- Validate delete events are handled for a resource with events: [all] in test config

**Comment by PrasadG193** [comment]
Sure @ShubhamSood1406, go ahead

**Comment by ShubhamSood1406** [comment]
> Sure @ShubhamSood1406, go ahead

Please @PrasadG193 can you explain what I have to do exactly as I am a beginner.

**Comment by PrasadG193** [comment]
@ShubhamSood1406 you can start with looking at test cases here: https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/delete/delete.go#L21
and here https://github.com/infracloudio/botkube/blob/develop/test/e2e/notifier/create/create.go#L48
And see how we create and validate an event. We need to do a similar thing for validating delete events, so to summarize, in the test, we need to - 
1. Create and delete a Pod resource 
2. Verify if the Slack and Webhook response

**Comment by pkosiec** [comment]
Hi, 
I'm closing this task as not valid anymore, as we switched to the E2E tests with real Slack and Kubernetes APIs and we have covered include/exclude namespaces feature in unit tests as well.
We can discuss new ideas under this epic: https://github.com/kubeshop/botkube/issues/589.

Thanks everyone involved!
